Change daemonset images name to match the built one

abdallahyas opened the pull request:
This patch change the antrea.yml image to match the built image.
This is needed because the latest antrea.yml uses different image
than the image name that was used when scripting the CI.

AbdYsn commented:
This PR should not affect the network operator CI, and only affects the sriov_antrea which passed the CI, and i think it is ready for review

Mmduh-483 commented on common/common_functions.sh:
```diff
@@ -502,6 +502,38 @@ function change_k8s_resource {
+    local file="$3"
+
+    let doc_num=0
+    changed="false"
```
`found` would serve as better variable for this case

Mmduh-483 commented on common/common_functions.sh:
```diff
@@ -502,6 +502,38 @@ function change_k8s_resource {
+        if [[ "$kind" == "DaemonSet" ]];then
+            name=$(yq r -d "$doc_num" $file metadata.name)
+            if [[ "$name" == "$daemonset_name" ]];then
+                let container_index=0
```
try to refactor it to be more readable
```
if [[ "$kind" == "DaemonSet" ]];then
            name=$(yq r -d "$doc_num" $file metadata.name)
            .............
```
to 
```
if [[ "$kind" != "DaemonSet" ]];then
            continue
...
```

Mmduh-483 commented on sriov_antrea/sriov_antrea_ci_start.sh:
```diff
@@ -75,10 +75,18 @@ EOF
+    change_image_name antrea/antrea-ubuntu antrea/antrea-ubuntu:ci
+
+    change_k8s_resource "DaemonSet" "antrea-agent" spec.template.spec.containers[*].image "antrea/antrea-ubuntu:ci" "$WORKSPACE/antrea-cni/build/yamls/antrea.yml"
+    change_k8s_resource "DaemonSet" "antrea-agent" spec.template.spec.initContainers[*].image "antrea/antrea-ubuntu:ci" "$WORKSPACE/antrea-cni/build/yamls/antrea.yml"
```
using `kustomize` may help you simplify this, WDYT?

abdallahyas committed:
Change antrea image to match the used image

This patch change the antrea.yml image to match the built image.
This is needed because the latest antrea.yml uses different image
than the image name that was used when scripting the CI.

AbdYsn commented:
@Mmduh-483 changed the patch so that it will be a simpler and more straight forward fix. please have a look

ykulazhenkov commented on sriov_antrea/sriov_antrea_ci_start.sh:
```diff
@@ -75,9 +75,13 @@ EOF
+
+    sed -i "s;\(^ *image\:\).*;\1 antrea/antrea-ubuntu;" $ARTIFACTS/antrea.yml
+ 
     if [[ -z "$(grep hw-offload $WORKSPACE/antrea-cni/build/yamls/antrea.yml)" ]];then
```
nit: I think it is more clear to do  grep in $ARTIFACTS folder. 
Maybe it will be better to use --hw-offload as pattern for grep?
if condition can be simplified to remove subshell usage
```
if ! grep -q  --hw-offload $ARTIFACTS/antrea-cni/build/yamls/antrea.yml; then
...
fi
```

AbdYsn commented:
@Mmduh-483 @ykulazhenkov can you have a look at https://github.com/vmware-tanzu/antrea/pull/2077
I am thinking of deprecating the project in this repo, since it is mainly hosted at the antrea repo itself and just keep it here as a reference. WDYT?

Mmduh-483 commented:
> @Mmduh-483 @ykulazhenkov can you have a look at [vmware-tanzu/antrea#2077](https://github.com/vmware-tanzu/antrea/pull/2077)


Won't this affect PR and daily CIs?

AbdYsn commented:
> > @Mmduh-483 @ykulazhenkov can you have a look at [vmware-tanzu/antrea#2077](https://github.com/vmware-tanzu/antrea/pull/2077)
> 
> Won't this affect PR and daily CIs?

PRs already use that, dailies should be modified to use that as well

AbdYsn commented:
closed since it was fixed in the antrea repo directly
